### PR TITLE
Expose current intersection index

### PR DESCRIPTION
**Pull request by LukasPaczos**
### Description

Closes https://github.com/mapbox/mapbox-navigation-android/issues/3623.

### Changelog

```
<changelog>Exposed current intersection index value in the `RouteProgress`.</changelog>
```

### CODE REVIEW

File: libnavigator/build.gradle [hunk 1]
```diff
@@ -58,6 +58,7 @@ dependencies {
 
     implementation dependenciesList.coroutinesCore
 
+    implementation(project(':libtesting-utils'))
```
Comment (RingerJK) [inline libnavigator/build.gradle:61]:
```suggestion
    testImplementation(project(':libtesting-utils'))
```

Comment (LukasPaczos) [reply libnavigator/build.gradle]: Good catch! Addressed.

File: libnavigator/src/test/resources/multileg_route.json [hunk 2]
```diff
@@ -0,0 +1 @@
...2.8,24.8,7.0,2.3,2.0,2.5,1.2,0.7,1.6,1.2,3.6,2.6,2.3,1.2,1.2,2.3,3.3,2.9,0.2],"congestion":["low","low","low","low","low","low","unknown","unknown","low","low","low","low","low","low","low","low","low","low","low","low","low","low","low","low","low","low","low","low","low","low","low","low","low","low","low","low","moderate","low","low","low","low","moderate","moderate","low","low","low","low","low","low","moderate","low","low","low","low","severe","heavy","heavy","low","low","low","low","low","low","low","low","low","low","low","low","unknown","unknown","unknown"]}}],"routeOptions":{"baseUrl":"https://api.mapbox.com","user":"mapbox","profile":"driving-traffic","coordinates":[[-77.1576396,38.7830304],[-77.1670888,38.7756155],[-77.1534183,38.7708948]],"alternatives":false,"language":"en","continue_straight":false,"roundabout_exits":false,"geometries":"polyline6","overview":"full","steps":true,"annotations":"congestion,distance,duration","voice_instructions":true,"banner_instructions":true,"voice_units":"imperial","access_token":"pk.eyJ1IjoiY2FmZXNpbGVuY2lvIiwiYSI6ImNrN2R3bnNxZDBkaTMzbm52bHJ2NmR4Z3YifQ.4ZUCu5Dqsyb_RfafMWe5tQ","uuid":"cka476dyl01qv7ak7ogvqdxgn"},"voiceLocale":"en-US"}
```
Comment (RingerJK) [inline libnavigator/src/test/resources/multileg_route.json:1]: nit could you format json for reading 🙇

File: libtesting-utils/src/main/java/com/mapbox/navigation/testing/FileUtils.kt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package com.mapbox.navigation.testing
+
+object FileUtils {
```
Comment (RingerJK) [inline libtesting-utils/src/main/java/com/mapbox/navigation/testing/FileUtils.kt:3]: ❤️

**Review by RingerJK — APPROVED** [review]

File: libnavigator/src/test/resources/multileg_route.json (hunk 2 above)
Comment (LukasPaczos) [reply libnavigator/src/test/resources/multileg_route.json:1]: Intentionally not formatted to decrease the size of the file and ultimately the size of the repo :)

Comment (RingerJK) [reply libnavigator/src/test/resources/multileg_route.json:1]: I've checked and it's 25 vs 50 kb :trollface: . I'm ok to save space))

Comment (LukasPaczos) [reply libnavigator/src/test/resources/multileg_route.json:1]: All I see is that it's twice the size :D

**Pull request merged**